packages/router/router.ts

```diff
@@ -3284,6 +3302,51 @@ export function createRouter(init: RouterInit): Router {
     );
   }
 
+  // single-patch implementation:
+  //   router.patchRoutes(parentId, children);
+  //
+  // batch-patch implementation:
+  //   router.patchRoutes((patch) => {
+  //     patch(parentId1, children1);
+  //     patch(parentId2, children2);
+  //     patch(parentId3, children3);
+  //   });
+  function patchRoutes(
+    routeIdOrCb:
+      | string
+      | null
+      | ((
+          patch: (r: string | null, children: AgnosticRouteObject[]) => void
+        ) => void),
+    children?: AgnosticRouteObject[]
+  ): void {
+    let isNonHMR = inFlightDataRoutes == null;
+    let routesToUse = inFlightDataRoutes || dataRoutes;
+    if (typeof routeIdOrCb === "function") {
+      routeIdOrCb((r, c) =>
+        patchRoutesImpl(r, c, routesToUse, manifest, mapRouteProperties)
+      );
+    } else if (typeof routeIdOrCb === "string" && children != null) {
+      patchRoutesImpl(
+        routeIdOrCb,
+        children,
+        routesToUse,
+        manifest,
+        mapRouteProperties
+      );
+    }
+
+    // If we are not in the middle of an HMR revalidation and we changed the
+    // routes, provide a new identity and trigger a reflow via `updateState`
+    // to re-run memoized `router.routes` dependencies.
+    // HMR will already update the identity and reflow when it lands
+    // `inFlightDataRoutes` in `completeNavigation`
+    if (isNonHMR) {
+      dataRoutes = [...dataRoutes];
+      updateState({});
+    }
```

Comment:
Update `router.routes` identity and reflow during eager `router.patchRoutes` flows